content/pro/openbb-copilot.md

```diff
@@ -112,24 +101,41 @@ To focus on certain widgets, click the "Add widgets as context" button on the wi
 
-## Querying your own data
+### Querying your own data
 
 OpenBB Copilot can also analyze and answer questions using files you provide, such as TXT, PDF, CSV, and XLSX files. To upload a file, simply drag and drop it into the OpenBB Copilot chatbox or click the paper clip icon.
 
 Once uploaded, OpenBB Copilot can use the data from these files to answer your questions. If it does, it will cite the specific files (and, for PDFs, the exact page) it referenced.
 
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/add_custom_data_chat_example.jpg" alt="add custom data openbb copilot" width="80%" height="80%" />
-
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_data_chat_example.jpg" alt="query custom data openbb copilot" width="80%" height="80%" />
-
 :::note
 
 OpenBB Copilot uses filenames to determine if a file is relevant to your query. For best results, use descriptive filenames.
 For example, if you have a technical report from Tesla released in 2024, a good filename would be `tsla_technical_report_2024.pdf`.
 
 :::
 
-## Querying external data
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/add_custom_data_chat_example.jpg" alt="add custom data openbb copilot" width="80%" height="80%" />
+
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_data_chat_example.jpg" alt="query custom data openbb copilot" width="80%" height="80%" />
+
+#### Adding custom data as a widget
+
+You can also add custom data as a widget to your dashboard. This allows you to analyze the data in the same way you would with any other widget.
+
+To do this, in the lower right corner of you dashboard, press the blue `+` button and select the "Add data" option. Then either drag and drop the file or click the "Browse for file(s)" button. Once you have uploaded the file(s), click the "Continue" button.
+
+| ![add custom data button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/add_custom_data_button.jpg) | ![add custom data upload openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/add_custom_data_uploaded_file.jpg) |
+|:-----------------------------------------------------------------------------------------:|:-----------------------------------------------------------------------------------------:|
+
+For each uploaded file, you can choose its widget name, and edit the metadata such as category, description, sub-category, and source. Once, satisfied, click the "Upload" button.
+
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_data_upload_button.jpg" alt="custom data upload openbb copilot" width="80%" height="80%" />
+
+Now, a widget with your data will be added to the dashboard.
+
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_data_widget_chat_example.jpg" alt="custom data widget openbb copilot" width="80%" height="80%" />
+
+### Direct retrieval
```

Comment:
It's not a perfect suggestions, but I would suggest something like this when talking about direct retrieval.
```suggestion
### Automatically using data sources available to Terminal Pro
```